### PR TITLE
Replace all drafts.csswg.org spec_url with w3c.github.io/csswg-drafts URLs

### PR DESCRIPTION
**Pull request by sideshowbarker**
Related browser-specs PR: https://github.com/w3c/browser-specs/issues/704

If that browser-specs PR gets held up or blocked on objection, then I suggest we add our own downstream special-casing in our own tooling — which I’m happy to work on as part of this PR.

I imagine we could update our spec-URL-check tooling to just rewrite (in memory) all w3c.github.io/csswg-drafts to their drafts.csswg.org equivalents before doing the browser-specs check.

In fact maybe we should just go ahead and do that now — rather than waiting for the browser-specs PR to be merge — because as of today (as I write this) drafts.w3.org was down again just 15 hours ago, for the 2nd time in two days.

So if we make this update in BCD now, we can immediately (as soon as the changes propogate to production) give MDN users reliable links to CSS specs in our Specifications tables in MDN.

### CODE REVIEW

File: test/linter/test-spec-urls.ts [hunk 1]
```diff
@@ -68,6 +68,7 @@ const processData = (data: CompatStatement, logger: Logger): void => {
     : [data.spec_url];
 
   for (const specURL of featureSpecURLs) {
```
Comment (sideshowbarker) [inline test/linter/test-spec-urls.ts:70]:
```suggestion
  for (let specURL of featureSpecURLs) {
```

File: test/linter/test-spec-urls.ts [hunk 2]
```diff
@@ -68,6 +68,7 @@ const processData = (data: CompatStatement, logger: Logger): void => {
...
+    specURL.replace('w3c.github.io/csswg-drafts', 'drafts.csswg.org');
```
Comment (sideshowbarker) [inline test/linter/test-spec-urls.ts:71]:
```suggestion
    specURL = specURL.replace('w3c.github.io/csswg-drafts', 'drafts.csswg.org');
```

**Comment by sideshowbarker** [comment]
> I imagine we could update our spec-URL-check tooling to just rewrite (in memory) all w3c.github.io/csswg-drafts to their drafts.csswg.org equivalents before doing the browser-specs check.

OK, I've now pushed an update here which does that. So CI is now all green here 🐍

**Review by queengooborg — APPROVED** [review]
Thank you for this PR, I am definitely down for replacing all of the links to drafts.csswg.org to something more reliable, and I think that we should do this ASAP.  (I have a sneaking suspicion that the browser-specs issue will be delayed as "the backend for drafts.csswg.org is being upgraded now"...)  Let's get it merged!

**Comment by sideshowbarker** [comment]
🎉💜 Thanks much

**Comment by Elchi3** [comment]
The test file says "When adding an exception here, provide a reason and indicate how the exception can be removed."
I think it would be nice also have a comment for this exception so that it is known why we derivate from browser-specs. 

(The idea was that we don't derivate from browser-specs, because that repo makes decisions for us about what is the canonical source, so every exception should be documented).

**Comment by Elchi3** [comment]
The downside of having exceptional `spec_urls` that are not in browser-specs is that we won't be able to gather spec metadata (like the title of a spec) for MDN spec tables. See https://github.com/mdn/content/issues/20772 for how this breaks MDN CSS spec tables.

**Pull request merged**